Petr's WGLC comments: protocol operation changes (#80)
ietf-wg-dnsop/draft-ietf-dnsop-structured-dns-error: draft-ietf-dnsop-structured-dns-error.md

## Client Generating Request {#client-request}

When generating a DNS query, a client that supports this specification

> SHOULD include the Structured DNS Error (SDE) option defined in {{SDE}}.

Reviewer: ```suggestion
SHOULD include the Structured DNS Error (SDE) option defined in {{SDE}}, unless instructed by local policy otherwise.
```

We need to provide valid exception for the SHOULD.

If the query contained the SDE EDNS option ({{client-request}}), and the
DNS server returns an EDE indicating blocking or modification of the response,
the DNS server SHOULD include additional detail in the EXTRA-TEXT field encoded

> as structured and machine-readable data.  If including the additional detail

Reviewer: ```suggestion
as structured and machine-readable data, unless configured otherwise.  If including the additional detail
```

would cause the response to exceed the EDNS0 size {{?RFC9715}} (and thus setting TC=1), it SHOULD be omitted.

If the SDE option was not present in the DNS request, the DNS server MUST NOT include

> structured JSON data and SHOULD convey the EXTRA-TEXT field as

Reviewer: ```suggestion
structured JSON data.  In such a case, the DNS server conveys the EXTRA-TEXT field as
```

Reviewer: We don't need the normative language for how RFC8914 is used. Normal processing would be followed per that spec.
